File: clients/php/src/CheFuAcademyException.php
<?php

declare(strict_types=1);

namespace CheFu\Academy;

final class CheFuAcademyException extends \RuntimeException
{
    public function __construct(string $message, private readonly ?int $statusCode = null)
    {
        parent::__construct($message, $statusCode ?? 0);
    }

    public function statusCode(): ?int
    {
        return $this->statusCode;
    }
}

File: clients/java/src/main/java/com/chefu/academy/CheFuAcademyException.java
package com.chefu.academy;

public final class CheFuAcademyException extends RuntimeException {
    private final int statusCode;

    public CheFuAcademyException(String message) {
        this(message, 0);
    }

    public CheFuAcademyException(String message, int statusCode) {
        super(message);
        this.statusCode = statusCode;
    }

    public int getStatusCode() {
        return statusCode;
    }
}
